spot-client/src/common/remote-control/P2PSignalingBase.js

```diff
@@ -0,0 +1,336 @@
+ */
+/**
+ * A peer to peer(direct) signaling channel used to send/receive remote control commands.
+ */
+export default class P2PSignalingBase {
+    /**
+     * Creates new instance.
+     *
+     * @param {P2PSignalingCallbacks} callbacks - The callbacks structure, see type description for more info.
+     * @param {P2PSignalingOptions} options - Extra options, see type description for more info.
+     */
+    constructor(callbacks, options) {
+        this._callbacks = callbacks;
+        this._options = options;
+
+        /**
+         * Stores active {@link PeerConnection}s mapped by remote address string.
+         *
+         * @type {Map<string, PeerConnection>}
+         * @private
+         */
+        this._peerConnections = new Map();
+    }
+
+    /**
+     * Find connection for remote address.
+     *
+     * @param {string} remoteAddress - The remote address string.
+     * @returns {?PeerConnection}
+     */
+    getConnectionForAddress(remoteAddress) {
+        return this._peerConnections.get(remoteAddress);
+    }
+
+    /**
+     * A convenience selector to find the first {@link PeerConnection}.
+     *
+     * @returns {?PeerConnection}
+     */
+    getFirstPeerConnection() {
+        return this._peerConnections.values().next().value;
+    }
+
+    /**
+     * Creates and initializes new {@link PeerConnection} instance for given remote address.
+     *
+     * @param {string} remoteAddress - The remote address string.
+     * @private
+     * @returns {PeerConnection}
+     * @throws an error if {@link PeerConnection} already exists for the given address.
+     */
+    _initPeerConnection(remoteAddress) {
+        if (this._peerConnections.get(remoteAddress)) {
+            throw new Error(`P2P signaling connection already exists for: ${remoteAddress}`);
+        }
+
+        const peerConnection = new PeerConnection(remoteAddress, this._options.iceServers);
+
+        this._peerConnections.set(remoteAddress, peerConnection);
+
+        peerConnection.addListener(
+            PeerConnection.DATA_CHANNEL_STATUS_CHANGED, (pc, isReady) => {
+                this._onReadyStateChanged(pc.remoteAddress, isReady);
+            });
+        peerConnection.addListener(
+            PeerConnection.DATA_CHANNEL_MSG_RECEIVED, (pc, data) => {
+                this._onDataChannelMessage(pc.remoteAddress, data);
+            });
+        peerConnection.addListener(
+            PeerConnection.ON_ICE_CANDIDATE, (pc, candidate) => {
+                // No 'candidate' means end of candidates
+                candidate && this._callbacks.onSendP2PMessage(
+                        pc.remoteAddress, { candidate: JSON.stringify(candidate) });
+            });
+
+        return peerConnection;
+    }
+
+    /**
+     * Checks if the first {@link PeerConnection} is active.
+     *
+     * @returns {boolean}
+     */
+    isReady() {
+        const first = this.getFirstPeerConnection();
```

Comment:
Why check the first specifically?

Reply:
It's been moved to the client version which only cares to have one PeerConnection.